src/libs/actions/Report.js

```diff
@@ -274,6 +274,45 @@ function addActions(reportID, text = '', file) {
         },
     ];
 
+    // Optimistically update the parent report action if the report is a thread
+    const report = ReportUtils.getReport(reportID);
+    if (ReportUtils.isThread(report)) {
+        const parentReportAction = ReportActionsUtils.getParentReportAction(report);
+        const childVisibleActionCount = parentReportAction.childVisibleActionCount ? parentReportAction.childVisibleActionCount + 1 : 1;
+        const childCommenterCount = parentReportAction.childCommenterCount ? parentReportAction.childCommenterCount : 1;
+        const childLastVisibleActionCreated = currentTime;
+        let childOldestFourEmails = parentReportAction.childOldestFourEmails;
+        if (!childOldestFourEmails) {
+            childOldestFourEmails = currentUserEmail;
+        } else {
+            const oldestFourEmails = childOldestFourEmails.split(',');
+            const index = _.findIndex(oldestFourEmails, (email) => email === currentUserEmail);
+            if (index !== -1) {
+                oldestFourEmails.splice(index, 1);
+            }
+            oldestFourEmails.push(currentUserEmail);
+            if (oldestFourEmails.length > 4) {
+                oldestFourEmails.splice(0, 1);
+            }
+            childOldestFourEmails = oldestFourEmails.join(',');
+        }
+
+        const optimisticParentReportAction = {
+            childVisibleActionCount,
+            childCommenterCount,
+            childLastVisibleActionCreated,
+            childOldestFourEmails,
```

Comment:
> When other user replied, it should show that user's avatar correctly even when offline, correct?

Yes.

> We need to know the logic that we used to update some parent report action fields here in backend to update optimisticData of parent report action when we add or delete a comment.

I'm a bit confused still about what logic you are referring to. Doesn't it just increment and decrement? And can't we just match what the API is returning?

Reply:
My apologies for my confusion, I'm just still not following what the exact problem is

Reply:
Checking the backend now

Reply:
Okay so on the backend it's a pretty complicated query. We don't actually ever modify those values directly, they are calculated based on the reportActions for a given report. 

So: 
childVisibleActionCount = number of visible reportActions (not whispers, moderator comments/actions, type_created, markedreimbursed, task_edited)
childCommenterCount = number of distinct accountIDs in all reportActions
childLastVisibleActionCreated = `MAX(created)` for all reportActions AKA the biggest timestamp
childOldestFourEmails = Sort all actions by created time -> Find the last 4 distinct accountIDs 


I'm not sure that's really helpful in this case unfortunately unless we want to also calculate all of that on the front end (which could be very slow for very large chat rooms). I'm thinking that we should manually assume that if you are offline and post a comment, that:
childVisibleActionCount has gone up by 1
childCommenterCount has gone up by 1 if you have not commented before
childLastVisibleActionCreated is the new comment with its generated timestamp
childOldestFourEmails is the same as it was if you were already in the list, otherwise the oldest one should be removed and you should be added (i'm not sure which one in the list is the oldest, but the position should be the same every time because of the query)